stat_var_renaming/stat_var_renaming.py

```diff
@@ -0,0 +1,631 @@
+  for num in range(1, row_constraints + 1): 
+    if not pd.isna(row[f"v{num}"]):
+      human_string += "_" + row[f"v{num}"]
+
+  # Postpend nmeasurement qualifier if it exists.
+  measurement_qualifier = row['measurementQualifier']
+  if not pd.isna(measurement_qualifier):
+    human_string = f"{human_string}_{measurement_qualifier}" 
+
+  # Postpend measurement denominator if it exists.
+  measurement_denominator = row['measurementDenominator']
+  if not pd.isna(measurement_denominator):
+    # Special Case: PerCapita which is directly appended.
+    if measurement_denominator == "PerCapita":
+      human_string = f"{human_string}_PerCapita"
+    # MDoms that are properties (all lower case) are added as Per(Mdom).
+    elif measurement_denominator[0].islower():
+      human_string = (
+        f"{human_string}_Per{capitalizeFirst(measurement_denominator)}")
+    # Everything else is AsAFractionOf.
+    else:
+      human_string = f"{human_string}_AsAFractionOf{measurement_denominator}"
+  return human_string
+
+def ensure_no_overlapping_stat_vars(stat_vars):
+  """ Ensures no collisions between statistical variables.
+  
+  This function ensures that there are not two distinct statistical variables
+  (e.g. different constraint, values, ect) that have the same name. There may
+  be two rows in the dataframe with the same name that have differences that
+  do not result in different statistical variables. E.g. same properties
+  except for measurement method.
+
+  Args:
+    stat_vars -> Dataframe of stat_vars with human readable name generated.
+
+  Returns:
+    stat_vars dataframe with no duplicate rows by name.
+  
+  Raises:
+    Assertion error if two distinct statistical variables evaluate to the same
+    name.
+  """
+  # Consider the subset of statistical variables that have at least one overlap.
+  overlapping_stat_var_names = (
+    stat_vars[stat_vars.duplicated('HumanReadableName', keep=False)])
+
+  # Check to make sure that none of these are actually different stat vars.
+  properties_of_statistical_variables = 
+    ['populationType','measuredProp', 'measurementQualifier',
+    'measurementDenominator', 'statType', 'scalingFactor']
+  for c in range(1, MAX_CONSTRAINTS + 1):
+    properties_of_statistical_variables.append(f"orig_p{c}")
+    properties_of_statistical_variables.append(f"orig_v{c}")
+
+  # See if there are any duplicates outside of this set.
+  bad_overlaps = overlapping_stat_var_names[
+      overlapping_stat_var_names.duplicated(
+        properties_of_statistical_variables, keep=False) == False]
+
+  # Print out error if there are real collisions.
+  bad_overlaps = bad_overlaps.sort_values(['v1', 'v2', 'v3'])
+  bad_overlaps_str = ""
+  for index, row in bad_overlaps.iterrows():
+    bad_overlaps_str += row_to_stat_var_mcf(row) + ","
+  assert bad_overlaps.shape[0] == 0, "Duplicate StatVars!: " + bad_overlaps
+
+  # No issues so remove these duplicate names.
```

Comment:
Are there ever duplicates that is actually the same?

Reply:
Yup! There's a lot of them as the SQL query pulls non-stat var params

Reply:
Ok. I added a TODO for myself to update the SQL query.